step2/main.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  main.swift
+//  step2
+//
+//  Created by jiye Yi on 2022/11/07.
+//
+
+import Foundation
+
+class Person {
+    var name: String
+    var gender: String
+    var moneyOnHand: Int
+    var job: String
+    init(name: String, gender: String, moneyOnHand: Int, job: String) {
+        self.name = name
+        self.gender = gender
+        self.moneyOnHand = moneyOnHand
+        self.job = job
+    }
+    
+    func order(_ coffee: Coffee, at coffeeshop: Coffeeshop) {
+        
+        if let coffeePrice = coffeeshop.menuList[coffee] {
+            
+            if moneyOnHand >= coffeePrice {
+                coffeeshop.make(coffee, from: name)
+                moneyOnHand -= coffeePrice
+                coffeeshop.turnover += coffeePrice
+            } else {
+                print("잔액이 \(coffeePrice - moneyOnHand) 만큼 부족합니다.")
+            }
+        }
+    }
+}
+
+class Coffeeshop {
+    let menuList = [Coffee.americano: 4500, Coffee.latte: 5000, Coffee.coldbrew: 5000, Coffee.frappuccino: 6000]
+    var turnover: Int = Int()
+    var barista: Person
+    var orderedCoffee: Coffee?
+    var customer: String?
+    var pickUpTable = [Coffee]() {
+        
+        didSet {
+            if let name = customer {
+                if let orderedCoffee = orderedCoffee {
+                    print("\(name)님이 주문하신 \(orderedCoffee.rawValue)(이/가) 준비되었습니다. 픽업대에서 가져가주세요.")
+                }
+            }
```

Comment:
```suggestion
            if let name = customer, let orderedCoffee = orderedCoffee {
                print("\(name)님이 주문하신 \(orderedCoffee.rawValue)(이/가) 준비되었습니다. 픽업대에서 가져가주세요.")
            }
```

if let 바인딩이 연달아 있다면, 이런식으로도 작성을 합니다 ☺️